ElrondNetwork/elrond-go
# Requests to preferred peers

Integrated the preferred peers holder component inside resolvers so if any preferred peer in the target shard exists, it will be used first when sending requests (only if there is more than one peer to request). Also skipped the antiflood checks if the peer the messages were received from is a preferred one.

Testing procedure:
1. in `prefs.toml` add public keys of any observer/validator from the testnet. 
2. start the testnet and run it with regular and upgrade scenarios for some time.
3. check the logs and look for the following log lines:

- `network connection status` - should contain a preferred peers histogram that isn't empty
- `sending request to preferred peer` 
- `request peers histogram` - should contain a peers histogram and preferred peers must be part of it
